proto/streamlit/proto/PageProfile.proto

```diff
@@ -0,0 +1,47 @@
+/**
+* Copyright 2018-2022 Streamlit Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+syntax = "proto3";
+
+
+message PageProfile {
+  repeated Command commands = 1;
+  int64 exec_time = 2;
+  int64 prep_time = 3;
+  repeated string config = 5;
+  string uncaught_exception = 6;
+  repeated string attributions = 7;
+}
+
+// The field names are used as part of the event json sent
+// to Segment. Since there is a 32kb limit for the event
+// size we are using short names to optimize for the size.
+message Argument {
+  // The keyword of the argument:
+  string k = 1;
```

Comment:
Was this done so that the metric_util.py code is shorter and easier to read?

Reply:
The comment above explains this